main.go

```diff
@@ -6,224 +6,43 @@
-			relModDir := *getModDir
-			modDir, err := filepath.Abs(relModDir)
-			if err != nil {
-				return errors.Wrap(err, "abs")
-			}
-			defer func() {
-				if err == nil {
-					// Leave tmp files on error for debug purposes.
-					if cerr := cleanGoGetTmpFiles(modDir); cerr != nil {
-						logger.Println("cannot clean tmp files", err)
-					}
-				}
-			}()
-
-			cfg := getConfig{
-				runner:    r,
-				modDir:    modDir,
-				relModDir: relModDir,
-				name:      *getName,
-				rename:    *getRename,
-				verbose:   *verbose,
-				link:      *getLink,
-			}
-
-			if err := get(ctx, logger, cfg, target); err != nil {
-				return errors.Wrap(err, "get")
-			}
-
-			pkgs, err := bingo.ListPinnedMainPackages(logger, modDir, true)
-			if err != nil {
-				return errors.Wrap(err, "list pinned")
-			}
-			if len(pkgs) == 0 {
-				return bingo.RemoveHelpers(modDir)
-			}
-			return bingo.GenHelpers(relModDir, version.Version, pkgs)
-		}
-	case "list":
-		listFlags.SetOutput(os.Stdout)
-		if err := listFlags.Parse(flags.Args()[1:]); err != nil {
-			exitOnUsageError(flags.Usage, "Failed to parse flags for get command:", err)
-		}
-
-		if !*verbose && *listVerbose {
-			*verbose = true
-		}
-
-		if *listModDir == "" {
-			exitOnUsageError(flags.Usage, "'moddir' flag cannot be empty")
-		}
-
-		if listFlags.NArg() > 1 {
-			exitOnUsageError(flags.Usage, "Too many arguments; only one binary/package or no argument is expected ")
-		}
-
-		target := listFlags.Arg(0)
-		cmdFunc = func(ctx context.Context, r *runner.Runner) error {
-			modDir, err := filepath.Abs(*listModDir)
-			if err != nil {
-				return errors.Wrap(err, "abs")
-			}
-			pkgs, err := bingo.ListPinnedMainPackages(logger, modDir, false)
-			if err != nil {
-				return err
-			}
-
-			bingo.SortRenderables(pkgs)
-			return pkgs.PrintTab(target, os.Stdout)
-		}
-	case "version":
-		cmdFunc = func(ctx context.Context, r *runner.Runner) error {
-			_, err := fmt.Fprintln(os.Stdout, version.Version)
-			return err
-		}
-	default:
-		exitOnUsageError(flags.Usage, "No such command", flags.Arg(0))
-	}
-
-	g := &run.Group{}
-	g.Add(run.SignalHandler(context.Background(), syscall.SIGINT, syscall.SIGTERM))
-
-	// Command run actor.
-	{
-		ctx, cancel := context.WithCancel(context.Background())
-		g.Add(func() error {
-			r, err := runner.NewRunner(ctx, logger, *getInsecure, *goCmd)
-			if err != nil {
-				return err
-			}
-
-			if *verbose {
-				r.Verbose()
-			}
-			return cmdFunc(ctx, r)
-		}, func(error) {
-			cancel()
-		})
-	}
-	if err := g.Run(); err != nil {
-		if *verbose {
-			// Use %+v for github.com/pkg/errors error to print with stack.
-			logger.Fatalf("Error: %+v", errors.Wrapf(err, "%s command failed", flags.Arg(0)))
-		}
-		logger.Fatalf("Error: %v", errors.Wrapf(err, "%s command failed", flags.Arg(0)))
+	rootCmd := NewBingoCommand(logger)
+	err := rootCmd.Execute()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
 	}
+	return
```

Comment:
nit: Do we need this? (no need for return in main)

Reply:
Yes, I'll delete him